Initial cut of the service-worker-libraries page

jeffposnick opened the pull request:
Attn: @addyosmani @wibblymat @PaulKinlan @tjsavage @jpmedley @beaufortfrancois 

Here's an initial cut of a landing page for `sw-precache` and `sw-toolbox`, under `/web/tools/service-worker-libraries`, implemented in the material design branch.

There's a decent amount of fine tuning to be done, but I wanted to get something out there to get the ball rolling. I tried to match the layout and tone of the sibling pages.

Here's what the first 90% of the page looks like, for those who can't run this PR locally:

![image](https://cloud.githubusercontent.com/assets/1749548/10498406/2e290588-7299-11e5-96e9-6806d961a452.png)

jeffposnick commented:
(`service-worker-libraries/thumb.png` is an ugly placeholder thumbnail image. I wish we have a service worker logo!)

PaulKinlan commented:
cc @pbakaus

pbakaus commented:
LGTM. You could change thumb.png to include the actual title as well, like all the other thumbs, to use the space for something, but not a big deal.

gauntface commented on src/content/en/tools/service-worker-libraries/index.markdown:
```diff
@@ -0,0 +1,138 @@
+title: "Service Worker Libraries"
+order: 6
+feedName: sw-toolbox Releases
+feedURL: https://github.com/googlechrome/sw-toolbox/releases.atom
```
@petele @PaulKinlan Do we want feeds like this pointing from WF to other pages?

petele replied:
I think this is fine, we've got it for Web Starter Kit and the Polymer Starter Kit, so I'm ok with it.

gauntface commented:
Only concern with this PR is that the thumb.png is nothing like the others and is going to look pretty dodgy compared to the others.

Anyone have any ideas / suggestions?

petele commented:
I'm OK (and support) this going into tools. 

Some minor feedback:
* I'd like to see a single sentence description under each header describing what the purpose of the library is. Imagine you arrive on the page, you want to know what the library does, not just the features.
* Instead of three cols for the features, since there are only 4, let's do 2 cols so things are balanced.

Thanks!
PEte

addyosmani commented:
@jeffposnick: This is totally awesome. Thanks for working on a draft so quickly. Do you want us to request designer assistance for the SW libraries logo? We might be able to get something churned out that's relatively simple. Fwiw, I agree with Pete on a brief description below each library. It should be instantly clear what they achieve without having to read the individual icons, despite those being useful for more granular feature info.
 
@petele: I'd like to try getting this page announced/up in time for CDS. We'll likely want to populate it with more content before then but there's no reason the current sw-precache/toolbox version couldn't go up sooner.

jpmedley commented:
Looking at this now.

jpmedley commented on src/content/en/tools/service-worker-libraries/index.markdown:
```diff
@@ -0,0 +1,138 @@
+      <h3 class="mdl-typography--title">Stay Fresh</h3>
+      <p>
+        Changes in your build update the service worker script. Users get updates without having to
+        manually version your content or caches.
```
"Users get updates, but you don't have to manually version your content or caches." The meaning of your sentence is that users are manually version content and caches.

jeffposnick replied:
Changed.

jpmedley commented on src/content/en/tools/service-worker-libraries/index.markdown:
```diff
@@ -0,0 +1,138 @@
+      <p>
+        Your static resources are served
+        <a href="https://jakearchibald.com/2014/offline-cookbook/#cache-falling-back-to-network">cache-first</a>,
+        whether or not there's a network available.
```
Since there are other caching patterns, perhaps the pattern used by this one shouldn't be buried in the text.

jeffposnick replied:
Cache-first is the only one used by `sw-precache`.

I added a link to the other patterns that `sw-toolbox` exposes to the new intro section.

jpmedley commented on src/content/en/tools/service-worker-libraries/index.markdown:
```diff
@@ -0,0 +1,138 @@
+      </div>
+      <h3 class="mdl-typography--title">Runtime Caching</h3>
+      <p>
+        Cache large or infrequently used resources, like images, at runtime, when they're
```
Is there an extra space after 'resources,'?

jeffposnick replied:
Nope, that's just a single space.

jeffposnick commented:
Thanks for the feedback, all!

I've switched to two columns via `mdl-cell--6-col`, which I'm assuming the correct way to do that.

I've added "Service Worker Libraries" to `thumb.png`, and while it's slightly more passable, getting some design help would be much appreciated, @addyosmani 

I've added one sentence intros to each section, so PTAL at that new text, @jpmedley and co.

petele commented:
I've merged and will do a push most likely tomorrow (Friday)
